src/main/java/org/candlepin/resteasy/filter/StoreFactory.java

```diff
@@ -165,15 +166,24 @@ public ConsumerStore(ConsumerCurator consumerCurator,
 
         @Override
         public Consumer lookup(final String consumerUuid) {
-            final Consumer byUuid = consumerCurator.findByUuid(consumerUuid);
-            if (byUuid != null) {
-                return byUuid;
+            Consumer consumer = null;
+            try {
+                consumer = consumerCurator.findByUuid(consumerUuid);
             }
+            catch (EntityNotFoundException e) {
+                // Intentionally ignoring exception
+            }
+
+            if (consumer != null) {
```

Comment:
Just a nitpick decide if you want to fix it. If we return the consumer in try block we no longer need to check if it is null.

Reply:
We won't be able to return the Consumer returned from `consumerCurator.findByUuid` right away because it is possible that a null Consumer is returned from this method, and we must check if there is a deleted consumer with that UUID by calling the `deletedConsumerCurator.countByConsumerUuid` method. This is done in the `wasDeleted` private method. This determines if we return a 404 response status code or a 410 response status code.

Reply:
yes, that true then it should be good